identity-apps-core/components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/listner/AppPortalRoleManagementListener.java

```diff
@@ -0,0 +1,160 @@
+
+        if (isAdministratorRole(roleId, tenantDomain)) {
+            throw new IdentityRoleManagementException(INVALID_REQUEST.getCode(),
+                "Deleting the 'Administrator' role belongs to the 'Console' application is not allowed.");
+        }
+    }
+
+    @Override
+    public void preUpdateUserListOfRole(String roleId, List<String> newUserIDList, List<String> deletedUserIDList,
+                                        String tenantDomain) throws IdentityRoleManagementException {
+
+        if (deletedUserIDList == null || !isAdministratorRole(roleId, tenantDomain)) {
+            return;
+        }
+
+        String adminUserId;
+        try {
+            UserRealm userRealm =
+                (UserRealm) PrivilegedCarbonContext.getThreadLocalCarbonContext().getUserRealm();
+            String adminUsername = userRealm.getRealmConfiguration().getAdminUserName();
+
+            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            adminUserId = ((AbstractUserStoreManager) userStoreManager).getUserIDFromUserName(adminUsername);
+        } catch (UserStoreException e) {
+            throw new IdentityRoleManagementException("Failed to retrieve user id of the tenant admin.", e);
+        }
+
+        if (deletedUserIDList.contains(adminUserId)) {
+            throw new IdentityRoleManagementException(INVALID_REQUEST.getCode(),
+                "Deleting the tenant admin from 'Administrator' role belongs to the 'Console' application is " +
+                    "not allowed.");
+        }
+    }
+
+    @Override
+    public void preUpdatePermissionsForRole(String roleId, List<Permission> addedPermissions,
+                                            List<Permission> deletedPermissions, String audience, String audienceId,
+                                            String tenantDomain) throws IdentityRoleManagementException {
+
+        if (isAdministratorRole(roleId, tenantDomain)) {
+            throw new IdentityRoleManagementException(INVALID_REQUEST.getCode(),
+                "Updating permissions of the 'Administrator' role belongs to the 'Console' application is not " +
+                    "allowed.");
+        }
+    }
+
+    private boolean isAdministratorRole(String roleId, String tenantDomain) throws IdentityRoleManagementException {
+
+        Role role = AppsCommonDataHolder.getInstance().getRoleManagementServiceV2().getRole(roleId, tenantDomain);
+        if (role == null || !StringUtils.equals(ADMINISTRATOR, role.getName())
+            || role.getAssociatedApplications() == null || role.getAssociatedApplications().isEmpty()) {
+            return false;
+        }
+
+        for (AssociatedApplication associatedApplication : role.getAssociatedApplications()) {
+            String applicationName = associatedApplication.getName();
+            if (StringUtils.isBlank(applicationName)) {
```

Comment:
Can we have an app without a name which is bound for an application audience role

Reply:
It actually have a name, but in the role.getAssociatedApplications() we only get the Ids and name come as null.